scripts/devaccess/devaccess_wrap.py

```diff
@@ -0,0 +1,454 @@
+        # create list of allowed commands for the user
+        for user in perm_dict['users']:
+            if user['username'] == self._user:
+                allowed_roles = user['roles']
+                # the 'ALL' group applies to everyone
+                allowed_roles.append('ALL')
+        logging.debug("user: %s roles: %s", self._user, str(allowed_roles))
+
+        commands = []
+        # get list of allowed commands for each role
+        for role in perm_dict['roles']:
+            if role['name'] in allowed_roles:
+                commands.extend(role['commands'])
+
+        logging.debug("user: %s commands: %s", self._user, str(commands))
+        return commands
+
+    @staticmethod
+    def whitelisted_command_list():
+        ''' Dict with key of whitelisted commmands mapped to their
+            actual implementation.
+            The commands should be in 'normalized' output style from OCCmd.
+        '''
+        command_dict = {}
+
+        ### commands for everyone (the 'ALL' group)
+        command_dict['oc get nodes -ndefault'] = WhitelistedCommands.oc_get_nodes
+        command_dict['oc get nodes -ndefault -ojson'] = WhitelistedCommands.oc_get_nodes
+        command_dict['rpm -qa'] = WhitelistedCommands.rpm_qa
+
+        return command_dict
+
+    def parse_args(self):
+        ''' Parse command line arguments passed in through the
+            SSH_ORIGINAL_COMMAND environment variable when READ_SSH is a
+            param.
+            Also handle when run manually. '''
+        args = None
+        user = None
+        read_ssh_env = False
+
+        # authorized_keys will force direct our command/argv to be
+        # 'devaccess_wrap READ_SSH <username>' with the original params stored
+        # in SSH_ORIGINAL_COMMAND
+        if "READ_SSH" in sys.argv:
+            read_ssh_env = True
+
+        if read_ssh_env:
+            cmd = os.environ.get("SSH_ORIGINAL_COMMAND", "")
+
+            # SSH_ORIGINAL_COMMAND will include the whole command
+            # as a continuous string
+            args = cmd
+
+            # The user's authorized_keys will force the command run to be:
+            # <path to devaccess_wrap> READ_SSH <username>
+            # Save the username for permission lookups later.
+            user = sys.argv[2]
+        else:
+            # not being launched from ssh authorized_keys, so
+            # drop the devaccess_wrap from the command
+            args = " ".join(sys.argv[1:])
+            user = os.getlogin()
+
+        self._args = args
+        self._user = user
+
+        if self._args.startswith('oc'):
+            self._oc_cmd = OCCmd(self._args)
+
+    def setup_logging(self):
+        ''' Configure logging '''
+
+        if os.environ.has_key("DEVACC_DEBUG") or self._debug is True:
+            log_level = logging.DEBUG
+        else:
+            log_level = logging.INFO
+
+        logging.basicConfig(filename=self._config['logfile_path'],
```

Comment:
Yes. The way I see it, the role that installs this RPM (openshift-tools-scripts-devaccess) will have to place a file with correct permissions and populate the config file pointing to the file.